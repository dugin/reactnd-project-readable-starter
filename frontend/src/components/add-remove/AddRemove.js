import React, {Component} from 'react';
import IosPlus from 'react-icons/lib/io/ios-plus';
import IosMinus from 'react-icons/lib/io/ios-minus';
import './AddRemove.css';
import PropTypes from 'prop-types';
import {voteType} from "../../utils/constants";

class AddRemove extends Component {

    constructor(props) {
        super(props);

        this.state = {quantity: props.amount || 1};
    }

    onAdd = () => {
        this.setState(state => state.quantity++, () => this.props.onAddOrRemove(voteType.upVote, this.state.quantity));
    };

    onRemove = () => {
        this.setState(state => state.quantity--, () => this.props.onAddOrRemove(voteType.downVote, this.state.quantity));
    };

    render() {

        return (
            <div className="AddRemove">
                <a onClick={this.onRemove} className="AddRemove__remove--btn"><IosMinus color="#39b54a" size={25}/></a>
                <a onClick={this.onAdd} className="AddRemove__add--btn"><IosPlus color="#39b54a" size={25}/></a>
            </div>
        )
    }
}

AddRemove.propTypes = {
    amount: PropTypes.number,
    onAddOrRemove: PropTypes.func.isRequired
};

export default AddRemove;