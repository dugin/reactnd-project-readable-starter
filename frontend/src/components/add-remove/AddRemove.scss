@import '../../variables';

.AddRemove {
  display: flex;
  justify-content: center;
  margin-top: 5px;

  a {
    color: map-get($theme-colors, success);
  }

  &__remove--btn {
    margin-right: 10px;
  }

}