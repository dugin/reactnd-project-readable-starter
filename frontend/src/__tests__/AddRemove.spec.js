import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';
import AddRemove from "../components/add-remove/AddRemove";

describe('AddRemove Component', () => {

    const addRemove = shallow(<AddRemove onAddOrRemove={jest.fn()} amount={1}/>);

    it('renders properly', () => {

        expect(toJson(addRemove)).toMatchSnapshot();

    });
});