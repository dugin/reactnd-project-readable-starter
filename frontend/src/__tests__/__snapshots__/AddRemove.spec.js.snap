// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddRemove Component renders properly 1`] = `
<div
  className="AddRemove"
>
  <a
    className="AddRemove__remove--btn"
    onClick={[Function]}
  >
    <IoIosMinus
      color="#39b54a"
      size={25}
    />
  </a>
  <a
    className="AddRemove__add--btn"
    onClick={[Function]}
  >
    <IoIosPlus
      color="#39b54a"
      size={25}
    />
  </a>
</div>
`;
